packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -7,16 +7,677 @@
+
+const TransitionShortHydrationLanePriority: LanePriority = 8;
+export const TransitionShortLanePriority: LanePriority = 7;
+
+const TransitionLongHydrationLanePriority: LanePriority = 6;
+export const TransitionLongLanePriority: LanePriority = 5;
+
+const SelectiveHydrationLanePriority: LanePriority = 4;
+
+const IdleHydrationLanePriority: LanePriority = 3;
+const IdleLanePriority: LanePriority = 2;
+
+const OffscreenLanePriority: LanePriority = 1;
+
+const NoLanePriority: LanePriority = 0;
+
+const TotalLanes = 30;
+
+export const NoLanes: Lanes = /*                        */ 0b000000000000000000000000000000;
+export const NoLane: Lane = /*                          */ 0b000000000000000000000000000000;
+
+export const SyncLane: Lane = /*                        */ 0b000000000000000000000000000001;
+const SyncUpdateRangeEnd = 1;
+export const SyncBatchedLane: Lane = /*                 */ 0b000000000000000000000000000010;
+const SyncBatchedUpdateRangeEnd = 2;
+
+export const InputDiscreteHydrationLane: Lane = /*      */ 0b000000000000000000000000000100;
+const InputDiscreteLanes: Lanes = /*                    */ 0b000000000000000000000000011100;
+const InputDiscreteUpdateRangeStart = 3;
+const InputDiscreteUpdateRangeEnd = 5;
+
+const InputContinuousHydrationLane: Lane = /*           */ 0b000000000000000000000000100000;
+const InputContinuousLanes: Lanes = /*                  */ 0b000000000000000000000011100000;
+const InputContinuousUpdateRangeStart = 6;
+const InputContinuousUpdateRangeEnd = 8;
+
+export const DefaultHydrationLane: Lane = /*            */ 0b000000000000000000000100000000;
+const DefaultLanes: Lanes = /*                          */ 0b000000000000000011111100000000;
+const DefaultUpdateRangeStart = 9;
+const DefaultUpdateRangeEnd = 14;
+
+const TransitionShortHydrationLane: Lane = /*           */ 0b000000000000000100000000000000;
+const TransitionShortLanes: Lanes = /*                  */ 0b000000000011111100000000000000;
+const TransitionShortUpdateRangeStart = 15;
+const TransitionShortUpdateRangeEnd = 20;
+
+const TransitionLongHydrationLane: Lane = /*            */ 0b000000000100000000000000000000;
+const TransitionLongLanes: Lanes = /*                   */ 0b000001111100000000000000000000;
+const TransitionLongUpdateRangeStart = 21;
+const TransitionLongUpdateRangeEnd = 25;
+
+export const SelectiveHydrationLane: Lane = /*          */ 0b000010000000000000000000000000;
+const SelectiveHydrationRangeEnd = 26;
+
+// Includes all non-Idle updates
+const UpdateRangeEnd = 26;
+const NonIdleLanes = /*                                 */ 0b000011111111111111111111111111;
+
+export const IdleHydrationLane: Lane = /*               */ 0b000100000000000000000000000000;
+const IdleLanes: Lanes = /*                             */ 0b011100000000000000000000000000;
+const IdleUpdateRangeStart = 27;
+const IdleUpdateRangeEnd = 29;
+
+export const OffscreenLane: Lane = /*                   */ 0b100000000000000000000000000000;
```

Comment:
"shelf"? 🤨 I'd be interested in hearing the reasoning behind this! 😆 Offscreen seemed pretty clear.

Reply:
Seb thinks that having an Offscreen component when something is not off screen is confusing. He is worried that people might misunderstand the name and assume that you only need to use `Offscreen` component if the elements are offscreen, which might result in people not wrapping something with Offscreen until the elements are hidden.

Reply:
`<KeepAlive>`, a la Vue?

Reply:
I think "shelf" might be a bit of a colloquialism that isn't very intuitive for a lot of people.

Isn't `Offscreen` always for things that might be **on or off** the screen? (e.g. list items that might be scrolled out of viewports, tab containers that might be hidden)

Is the concern that you'd always have an `Offscreen` wrapper around even the visible items in that case (since we don't support reparenting)?

Maybe something like `Conditional` would be more accurate/intuitive?

Reply:
It does seem related to the concept of `keep-alive`...

Reply:
Agree that a public API called `Offscreen` is confusing. Similar to when we were trying to think of a name for the `Suspense` component.

The naming considerations are slightly different for this lane, though. The thing that makes `Offscreen` special, aside from it being really low priority, is that we're allowed to not finish it, because the work at that level is not visible to the user.

https://github.com/facebook/react/blob/17448e1e7f9131215dfddfff1aeddf3812073dd9/packages/react-reconciler/src/ReactFiberWorkLoop.new.js#L1654-L1659

A better solution, though, might be to move `resetChildLanes` (the function in that snippet) to the switch statement inside `completeWork`. Then the only thing remarkable about the Offscreen is that it has really low priority.

Reply:
`<MaybeOrNotOffscreen />` 😆